### PR TITLE
Add `prefect.yaml` `pull` step generation based on `build_docker_image` `build` step 

### PR DESCRIPTION
**Pull request by serinamarie**
Creates a `pull` step when `build_docker_image` build step exists. 

If a `prefect_docker.deployments.steps.build_docker_image` step exists in the `build` section of the `prefect.yaml`, and no `pull` steps exist, we generate a `pull` step for the user. One of:
- `prefect.deployments.steps.git_clone` 
- `prefect.deployments.steps.set_working_directory`

depending on the user's `dockerfile` arg in the `build` step. 

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: src/prefect/cli/deploy.py [hunk 1]
```diff
@@ -721,10 +722,156 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
...
+        else:
+            if is_interactive():
+                if remote_url and confirm(
+                    "Would you like to pull your flow code from its remote"
+                    " repository when running your deployment?"
+                ):
+                    return await _generate_git_clone_pull_step(
+                        console, deploy_config, remote_url
+                    )
+                if not confirm("Do you copy your flow code in your Dockerfile?"):
```
Comment (desertaxle) [inline src/prefect/cli/deploy.py:866]:
```suggestion
                if not confirm("Does your Dockerfile have a line that copies the current working directory into your image?"):
```

File: src/prefect/cli/deploy.py [hunk 2]
```diff
@@ -721,10 +722,156 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
...
+                    exit_with_error(
+                        "Your flow code must be copied into your Docker image"
+                        " in order to run your deployment."
```
Comment (desertaxle) [inline src/prefect/cli/deploy.py:869]: Suggestion:

Your flow code must be copied into your Docker image to run your deployment.

To copy your flow code into your Docker image you can this line to your Dockerfile:

    COPY . /opt/prefect/

Comment (serinamarie) [reply src/prefect/cli/deploy.py]: #nodeadends

File: src/prefect/cli/deploy.py [hunk 3]
```diff
@@ -726,10 +727,160 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
...
+
+    return None
+
+
 async def _generate_default_pull_action(
-    console: Console, deploy_config: Dict, ci: bool = False
+    console: Console, deploy_config: Dict, actions: List[Dict], ci: bool = False
 ):
     remote_url = _get_git_remote_origin_url()
+    build_docker_image_step = await _check_for_build_docker_image_step(actions["build"])
```
Comment (desertaxle) [inline src/prefect/cli/deploy.py:854]: The `build` action could also be specific to this deploy config.

```suggestion
    build_docker_image_step = await _check_for_build_docker_image_step(deploy_config.get("build") or actions["build"])
```

File: src/prefect/cli/deploy.py [hunk 4]
```diff
@@ -726,10 +727,160 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
...
+):
+    pull_step = {}
+    if auto:
+        pull_step["path"] = "/opt/prefect/{{ name }}"
+    else:
+        pull_step["path"] = prompt(
+            "What is the path to your flow code in your Dockerfile?",
+            default="/opt/prefect/{{ name }}",
+            console=console,
+        )
```
Comment (desertaxle) [inline src/prefect/cli/deploy.py:827]: The `{{ name }}` placeholder won't template correctly in most cases, so using the current directory name is safer. It aligns with the behavior in `prefect-docker` as well.

```suggestion
    dir_name = os.path.basename(os.getcwd())
    if auto:
        pull_step["directory"] = f"/opt/prefect/{dir_name}"
    else:
        pull_step["directory"] = prompt(
            "What is the path to your flow code in your Dockerfile?",
            default=f"/opt/prefect/{dir_name}",
            console=console,
        )
```

**Review by desertaxle — APPROVED** [review]
🚀 LGTM!

**Pull request merged**